### ISSUE TITLE
Enum dunder method `_generate_next_value_` not allowed

### ISSUE THREAD

**Issue by n4nn31355** [pylint-dev/pylint#10435]
### Bug description

Enum dunder method `_generate_next_value_` is missing in list of allowed methods

```python
# Bad or misspelled dunder method name _generate_next_value_. Pylint(W3201:bad-dunder-name)
class PylintDunderEnum(Enum):
    @staticmethod
    def _generate_next_value_(
        name: str, _start: int, _count: int, _last_values: list[Any]
    ) -> Any:
        return name
```

### Configuration

```ini

```

### Command used

```shell
pylint .\enums.py
```

### Pylint output

```python
.\enums.py:4:4: W3201: Bad or misspelled dunder method name _generate_next_value_. (bad-dunder-name)
```

### Expected behavior

No error when Enum dunder method `_generate_next_value_` is used

### Pylint version

```shell
pylint 3.3.7
astroid 3.3.10
Python 3.11.9 (tags/v3.11.9:de54cf5, Apr  2 2024, 10:12:12) [MSC v.1938 64 bit (AMD64)]
```

### Additional dependencies

```python

```

**Comment by mbyrnepr2** [comment]
A couple more:
[\_add_alias\_](https://docs.python.org/3.15/library/enum.html#enum.EnumType._add_alias_)
[\_add_value_alias\_](https://docs.python.org/3.15/library/enum.html#enum.EnumType._add_value_alias_)
[\_missing\_](https://docs.python.org/3/library/enum.html#enum.Enum._missing_)
[\_numeric_repr\_](_numeric_repr_)
